server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -320,6 +323,19 @@ public short registerBrokerRecordVersion() {
         }
     }
 
+    public short registerControllerRecordVersion() {
+        if (isAtLeast(MetadataVersion.IBP_3_6_IV2)) {
```

Comment:
We are not using `IBP_3_7_IV0`, does this mean a old-version controller can register a controller with an new-version controller?

Reply:
Sorry, this was left over from when the feature was in `IBP_3_6_IV2` previously. Fixed now.